src/metadata/EntityMetadata.ts

```diff
@@ -861,6 +862,11 @@ export class EntityMetadata {
             }
         }
 
+        if (this.connection.driver instanceof SqliteDriver) {
+            // @todo - perhaps use this driver callback in all cases?
```

Comment:
At this point lets just abstract `buildTablePath()` for all drivers and pass `tableName`, `schema` and `database`. The function could be made optional also and default to using `tableName` by default.

Reply:
I didn't want to do this outright in my first PR because i'm still unclear the differences between `tableName` and `tablePath`.

I think it needs a bit more thought (and probably needs more consistent escaping strategies in certain places)

Reply:
The only difference should be that `tablePath` contains the full database/schema name, while `tableName` does not. I think drivers have also so far been responsible for escaping, but is there something else I'm missing here?

Reply:
the concept of `tablePath` vs `tableName` seems a little blurry throughout - it differs from here in the `EntityMetadata`, through the `Table` and `TableIndex` instances. Also, the same driver instance can expect differing values based on the platform (the `AbstractSqliteDriver/QueryRunner`). e.g. `sql.js` does not support tablePaths on indexes etc

Reply:
tl;dr; i'd rather not include cross-driver changes in this PR